### PR TITLE
Patch analysis

### PR DESCRIPTION
**Pull request by lrivera-oicr**
- New endpoint PATCH /studies/{studyId}/analysis/{analysisId}
- Send Kafka notification

### CODE REVIEW

File: song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceSender.java [hunk 1]
```diff
@@ -96,8 +96,17 @@ public Analysis suppress(String studyId, String id) {
...
+  public Analysis updateAnalysis(String studyId, String analysisId, JsonNode updateAnalysisRequest) {
+    Analysis analysis = internalAnalysisService.updateAnalysis(studyId, analysisId, updateAnalysisRequest);
+    sendAnalysisMessage(analysis, PUBLISHED, PUBLISH);
+    return analysis;
+  }
+
+  @Override
+  public Analysis patchUpdateAnalysis(String studyId, String analysisId, JsonNode patchUpdateAnalysisRequest) {
+    Analysis analysis = internalAnalysisService.patchUpdateAnalysis(studyId, analysisId, patchUpdateAnalysisRequest);
+    sendAnalysisMessage(analysis, PUBLISHED, PUBLISH);
```
Comment (caravinci) [inline song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceSender.java:108]: does this send the analysis in the response?

Comment (lrivera-oicr) [reply song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceSender.java:108]: this method returns a response. however is not used in the AnalysisController class.

File: song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceImpl.java [hunk 2]
```diff
@@ -168,6 +168,43 @@ public void updateAnalysis(
...
+  @Transactional
+  public Analysis patchUpdateAnalysis(
+          @NonNull String studyId,
+          @NonNull String analysisId,
+          @NonNull JsonNode patchUpdateAnalysisRequest) {
+    // Validate prerequisites
+    checkAnalysisAndStudyRelated(studyId, analysisId);
+
+    // Now that the request is validated, fetch the old analysis
+    val analysis = get(analysisId, true, true, false);
```
Comment (caravinci) [inline song-server/src/main/java/bio/overture/song/server/service/analysis/AnalysisServiceImpl.java:185]: `val` for consistency, eh? lol

**Review by caravinci — APPROVED** [review]
Looks good to me, let's try it out in dev! 🚀

**Pull request merged**